bppy/model/event_selection/simple_event_selection_strategy.py

```diff
@@ -37,10 +37,13 @@ def selectable_events(self, statements):
                     possible_events = {x for x in possible_events if x not in statement.get('block')}
         return possible_events
 
-    def select(self, statements):
+    def select(self, statements, external_events_queue=[]):
         selectable_events = self.selectable_events(statements)
         if selectable_events:
             return random.choice(tuple(selectable_events))
         else:
-            return None
+            if len(external_events_queue) > 0:
```

Comment:
This is the correct order of handling external events and internal events(checked with Guy).
1. finish the superstep 
2. Inject the next external event.
Nice Job!